src/test/java/org/apache/drillui/test/framework/testng/secure/login/LoginTest.java

```diff
@@ -16,23 +16,90 @@
+    assertEquals(  new AuthSteps().logOut().getLoginText(), "Log In", "Logout failed");
+  }
+
+  @Test(groups = {"functional"})
+  public void AdminsViewVsUsersViewInSecurityMode()  {
+    //Step 1. Run query for Admin1
+    Assert.assertEquals("COMPLETED",new AuthSteps().login(PropertiesConst.ADMIN_1_NAME,PropertiesConst.ADMIN_1_PASSWORD)
+            .navigateQuery()
+            .submitQuery("SELECT * FROM cp.`employee.json` LIMIT 20")
+            .getQueryStatus());
+    String Admin_1_QueryProfile = new QueryResultsSteps().getQueryProfile();
+    new AuthSteps().logOut();
+
+    //Step 2. Run query for Admin2
+    Assert.assertEquals("COMPLETED", new AuthSteps().login(PropertiesConst.ADMIN_2_NAME,PropertiesConst.ADMIN_2_PASSWORD)
+            .navigateQuery()
+            .submitQuery("SELECT * FROM cp.`employee.json` LIMIT 20")
+            .getQueryStatus());
+    String Admin_2_QueryProfile = new QueryResultsSteps().getQueryProfile();
+    new AuthSteps().logOut();
+
+    //Step 3. Run query for User1
+    Assert.assertEquals("COMPLETED", new AuthSteps().login(PropertiesConst.USER_1_NAME,PropertiesConst.USER_1_PASSWORD)
+            .navigateQuery()
+            .submitQuery("SELECT * FROM cp.`employee.json` LIMIT 20")
+            .getQueryStatus());
+    String User_1_QueryProfile = new QueryResultsSteps().getQueryProfile();
+    new AuthSteps().logOut();
+
+    //Step 4. Run query for User2
+    Assert.assertEquals("COMPLETED", new AuthSteps().login(PropertiesConst.USER_2_NAME,PropertiesConst.USER_2_PASSWORD)
+            .navigateQuery()
+            .submitQuery("SELECT * FROM cp.`employee.json` LIMIT 20")
+            .getQueryStatus());
+    String User_2_QueryProfile = new QueryResultsSteps().getQueryProfile();
+    new AuthSteps().logOut();
+
+    //Step 5. Checks that Admin1 can see query of all Admins/Users
+   Assert.assertTrue(new AuthSteps().login(PropertiesConst.ADMIN_1_NAME,PropertiesConst.ADMIN_1_PASSWORD)
+           .navigateProfiles()
+           .isQueryProfileINList(Admin_1_QueryProfile));
+    Assert.assertTrue(new ProfilesSteps().isQueryProfileINList(Admin_2_QueryProfile));
+    Assert.assertTrue(new ProfilesSteps().isQueryProfileINList(User_1_QueryProfile));
+    Assert.assertTrue(new ProfilesSteps().isQueryProfileINList(User_2_QueryProfile));
+    new AuthSteps().logOut();
+
+   //Step 6. Checks that User can see only his queries via assertTrue
+   Assert.assertTrue( new AuthSteps().login(PropertiesConst.USER_1_NAME,PropertiesConst.USER_1_PASSWORD)
+           .navigateProfiles()
+           .isQueryProfileINList(User_1_QueryProfile));
+
+   // and DO NOT SEE other queries via assertFalse
+   Assert.assertFalse(new ProfilesSteps().isQueryProfileINList(User_2_QueryProfile));
+   Assert.assertFalse(new ProfilesSteps().isQueryProfileINList(Admin_1_QueryProfile));
+   Assert.assertFalse(new ProfilesSteps().isQueryProfileINList(Admin_2_QueryProfile));
```

Comment:
Could you please add users verification for other profiles in the list under these four profiles?

Reply:
Done